src/InputFileResolver.ts

```diff
@@ -105,14 +105,15 @@ class PatternResolver {
       // Start the globbing task for the current descriptor
       const globbingTask = this.resolveGlobbingExpression(this.descriptor.pattern)
         .then(filePaths => filePaths.map(filePath => this.createInputFile(filePath)));
+     
+      // If there is a previous globbing expression, resolve that one as well
       if (this.previous) {
-        // If there is a previous globbing expression, resolve that one as well
         return Promise.all([this.previous.resolve(), globbingTask]).then(results => {
           const previousFiles = results[0];
           const currentFiles = results[1];
           // If this expression started with a '!', exclude current files
           if (this.ignore) {
-            return previousFiles.filter(previousFile => currentFiles.some(currentFile => previousFile.path !== currentFile.path));
+            return previousFiles.filter(previousFile => currentFiles.every(currentFile => previousFile.path !== currentFile.path));
```

Comment:
When an expression starts with a `!` (and needs to ignore all matching files), then every file resulting from the current globing expression should not match to the current expression.

For example:
* `previousFiles: ['file1', 'file2', 'expectedFile']`
* `currentFiles: ['file1', 'file2', 'somefile']`
* Expected result: `'expectedFile'`

For an empty array, `every` returns true. More info: https://developer.mozilla.org/nl/docs/Web/JavaScript/Reference/Global_Objects/Array/every